### ISSUE TITLE
Function type overloads in inconsistent state

### ISSUE THREAD

**Issue by letmaik** [microsoft/pyright#623]
**Describe the bug**
I'm not sure how to describe this properly but the screenshots below should speak volumes.

**Screenshots or Code**

![grafik](https://user-images.githubusercontent.com/530988/79366329-e8977d80-7f43-11ea-809e-a2d50f9e275d.png)

![grafik](https://user-images.githubusercontent.com/530988/79366349-f2b97c00-7f43-11ea-8a9f-5716877189ed.png)

![grafik](https://user-images.githubusercontent.com/530988/79366606-66f41f80-7f44-11ea-9067-90e474306a04.png)

Note I only changed the "capture" argument between the first two screenshots.
In the last screenshot I removed and re-added some code which seemed to reset pyright.
Basically, pyright seems to get into an inconsistent state sometimes.

```
from typing import Optional, Literal, overload

@overload
def run(*args: str, capture: Literal[True]=True) -> str:
    pass

@overload
def run(*args: str, capture: Literal[False]=False) -> None:
    pass

def run(*args: str, capture: bool=False) -> Optional[str]:
    # ...implementation...
    if capture:
        return 'foo'
    return None

foo = run('time', capture=True)
bar = run('time', capture=False)
```

**VS Code extension or command-line**
VS Code

**Additional context**

**Comment by erictraut** [comment]
That was a subtle bug, but I found it. Thanks for your great bug report! The fix will be in the next version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.1.32, which I just published.
